lib/util/mixins.js

```diff
@@ -42,34 +40,33 @@ module.exports = {
 		return options;
 	},
 
+	getQueueName ( messageType, prefix ) {
+		const parts = this.extract( messageType );
+
+		if ( parts instanceof Error ) {
+			return parts;
+		}
+
+		return `${prefix}.${parts.resource}-${parts.action}`;
+	},
+
 	getProducerOptions ( options, prefix ) {
 		if ( !options || !options.messageType ) {
 			return new Error( 'NULL messageType' );
 		}
 
 		// returns error if no msgType or if invalid
 		const parts = this.extract( options.messageType );
-		let returnData;
 
 		if ( parts instanceof Error ) {
-			returnData = parts;
-		} else {
-			// if no exchange, create exchange from msgType
-			const partExchange = options.exchange ||  parts.resource;
-			const exchange     = prefix + '.' + partExchange + '-exchange';
-
-			// should we override the exchange and messageType
-			const transformedOptions = {
-				exchange,
-				'messageType' : prefix + '.' + options.messageType
-			};
-
-			// strip off to persists transformed options
-			options = _.omit( options, 'messageType', 'exchange' );
-
-			returnData = _.extend( transformedOptions, options );
+			return parts;
 		}
-		return returnData;
+
+		// should we override the exchange and messageType
+		return Object.assign( {}, options, {
+			'exchange'    : options.exchange || `${prefix}.${parts.resource}-exchange`,
```

Comment:
This is different than was previously in there. I'm not sure if you meant to do this. I lie this way better, because if they pass in an `exchange` we shouldn't be prefixing and suffixing it, but the old logic did.